serve/mlc_serve/engine/staging_engine.py

```diff
@@ -192,14 +194,18 @@ def step(self) -> InferenceStepResult:
     def _is_ready_to_serve(self) -> bool:
         return self.worker_process is not None and self.worker_process.is_alive()
 
-    def _get_new_request_state(self, request: Request) -> RequestState:
+    def _get_new_request_state(self, request: Request) -> Optional[RequestState]:
```

Comment:
The return value is never `Optional`, is it?

Reply:
Ah yes this is a piece of bit rot from my previous attempt. Let me change that.